src/jshint.js

```diff
@@ -2562,11 +2549,45 @@ var JSHINT = (function() {
       return left;
     }
 
+    if (next.id === "+" || next.id === "++") {
+      warning("W007", that.right);
+    }
+
+    return that;
+  }, 130);
+  prefix("+", function(context) {
+    var next = state.tokens.next;
+    this.arity = "unary";
+    this.right = expression(context, 150);
+
+    if (next.id === "+" || next.id === "++") {
+      warning("W007", this.right);
+    }
+
+    return this;
+  });
+  infix("-", function(context, left, that) {
+    var next = state.tokens.next;
+    that.left = left;
+    that.right = expression(context, 130);
+
+    if (next.id === "-" || next.id === "--") {
+      warning("W006", that.right);
+    }
+
     return that;
   }, 130);
-  prefix("+", "num");
-  infix("-", "sub", 130);
-  prefix("-", "neg");
+  prefix("-", function(context) {
+    var next = state.tokens.next;
+    this.arity = "unary";
+    this.right = expression(context, 150);
+
+    if (next.id === "-" || next.id === "--") {
+      warning("W006", this.right);
+    }
+
+    return this;
+  });
```

Comment:
I'm so tempted to ask for some abstraction for all of these, but I think that I will just accept that sometimes it's nice to just have a clear view of the operations.